packages/commons-test/src/mockedPayloadForToken.ts

```diff
@@ -75,27 +75,32 @@ function createInternalPayload() {
   };
 }
 
-function createM2M_AdminPayload() {
+export const mockM2MAdminClientId = generateId();
+export const mockM2MAdminUserId: UserId = generateId();
+// ^ ID of the client and the admin user associated with the client.
+// Mocked and exported because in the M2M gateway we need to
+// validate the admin ID in the token against the adminId in the client.
+function createM2MAdminPayload() {
   return {
     iss: "dev.interop.pagopa.it",
     aud: "dev.interop.pagopa.it/ui",
-    userId: generateId(),
     exp: Math.floor(Date.now() / 1000) + 3600,
     nbf: Math.floor(Date.now() / 1000),
     iat: Math.floor(Date.now() / 1000),
     jti: "1bca86f5-e913-4fce-bc47-2803bde44d2b",
     role: "m2m-admin",
     organizationId: generateId(),
-    client_id: generateId(),
-    sub: generateId(),
+    client_id: mockM2MAdminClientId,
+    sub: mockM2MAdminClientId,
+    userId: mockM2MAdminUserId,
```

Comment:
```suggestion
    adminId: mockM2MAdminUserId,
```
in the jwt we opted for calling it `adminId` instead of `userId`

Reply:
The PR that changes `userId` to `adminId` in the token is still unmerged: https://github.com/pagopa/interop-be-monorepo/pull/1824
This one needs to be merged this week to do some performance tests, I'd say we leave userId and then the PR linked above changes it to `adminId` here as well.
Unless it gets merged before, in that case I'll do the fix here (tests should break, so there is not risk to forget)